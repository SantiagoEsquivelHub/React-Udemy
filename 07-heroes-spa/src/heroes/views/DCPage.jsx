import React from 'react'
import { HeroList } from '../components'

export const DCPage = () => {
  return (
    <div>
      <HeroList publisher='DC Comics' />
    </div>
  )
}
