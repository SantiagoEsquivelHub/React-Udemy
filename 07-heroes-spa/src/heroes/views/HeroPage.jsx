import { useParams, Navigate } from 'react-router-dom'
import { getHeroeById } from '../helpers';

export const HeroPage = () => {

  const { id } = useParams();

  const heroe = getHeroeById(id);

  if (!heroe) {
    return <Navigate to="/marvel"/>
  }

  return (
    <div>
      <h1>HeroPage</h1>
      <hr />
    </div>
  )
}
