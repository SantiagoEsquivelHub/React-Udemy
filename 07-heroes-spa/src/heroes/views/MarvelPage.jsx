import React from 'react'
import { HeroeList } from '../components'

export const MarvelPage = () => {
    return (
        <div>
            <HeroeList publisher='Marvel Comics' />
        </div>
    )
}
