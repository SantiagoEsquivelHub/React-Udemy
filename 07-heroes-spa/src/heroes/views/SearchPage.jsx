import { useLocation, useNavigate } from 'react-router-dom';
import { useForm } from '../../hooks/useForm';
import { HeroCard } from '../components'
import queryString from 'query-string';
import { getHeroesByName } from '../helpers';

export const SearchPage = () => {

  const navigate = useNavigate();
  const location = useLocation();
  const { q = '' } = queryString.parse(location.search);

  const heroes = getHeroesByName(q);

  const showSearch = (q.length === 0);
  const showError = (q.length > 0) && (heroes.length === 0);

  const { formState, onInputChange, handleReset, searchText } = useForm({
    searchText: q
  });

  const onSearchSubmit = (e) => {
    e.preventDefault();

    if (searchText.trim().length <= 1) return;

    navigate(`?q=${searchText}`);
    console.log({ q })
  }

  return (
    <div>

      <h4>Search</h4>
      <hr />

      <div className="row">
        <div className="col-5">
          <h4>Searching...</h4>
          <hr />
          <form onSubmit={onSearchSubmit} aria-label="form">
            <input
              type="text"
              placeholder='Search a heroe'
              className='form-control mb-2'
              name='searchText'
              autoComplete='off'
              value={searchText}
              onChange={onInputChange}
            />

            <button className='btn btn-outline-primary'>
              Search
            </button>
          </form>
        </div>

        <div className="col-7">
          <h4>Results</h4>
          <hr />

          <div className="alert alert-primary animate__fadeIn animate__animated" style={{ display: showSearch ? '' : 'none' }} aria-label="search_heroe">
            Search a heroe
          </div>

          <div className="alert alert-danger animate__fadeIn animate__animated" style={{ display: showError ? '' : 'none' }} aria-label="error_search_heroe">
            There's no heroe with <b>{q}</b>
          </div>


          {
            heroes.map(heroe => (
              <HeroCard key={heroe.id} {...heroe} />
            ))
          }

        </div>

      </div>
    </div >
  )
}
