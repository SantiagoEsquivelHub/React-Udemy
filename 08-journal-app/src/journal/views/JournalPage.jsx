import { Typography } from '@mui/material'
import React from 'react'
import { JournalLayout } from '../layout/JournalLayout'

export const JournalPage = () => {
  return (
    <JournalLayout>
      <Typography>dfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddfdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddfdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddfdfasdfasdfasdsdfasdfasdfasdfasdfasdfasdfasdfasddddddddddddddddddddddf</Typography>

      {/* NothingSelected */} 

      {/* NoteView */} 

    </JournalLayout>
  )
}
