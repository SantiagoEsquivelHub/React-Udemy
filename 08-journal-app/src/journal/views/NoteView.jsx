import { useEffect, useMemo } from "react"
import { Button, Grid, TextField, Typography } from "@mui/material"
import { useDispatch, useSelector } from "react-redux"
import Swal from "sweetalert2"
import moment from "moment/moment"
import { SaveOutlined } from "@mui/icons-material"
import { useForm } from "../../hooks/useForm"
import { setActiveNote } from "../../store/journal/journalSlice"
import { startSavingNotes } from "../../store/journal/thunks"
import { ImageGallery } from "../components"
import "sweetalert2/dist/sweetalert2.css";

export const NoteView = () => {

    const dispatch = useDispatch();

    const { active: note, savedMessage, isSaving } = useSelector(state => state.journal)
    const { title, body, date, onInputChange, formState } = useForm(note);

    const dateString = useMemo(() => {
        const newDate = new Date(date);
        return newDate.toUTCString();
    }, [date])

    useEffect(() => {
        dispatch(setActiveNote(formState));
    }, [formState])

    useEffect(() => {
        if (savedMessage.length > 0) {
            Swal.fire('Nota actualizada!', savedMessage, 'success')
        }
    }, [savedMessage])

    const onSaveNote = () => {
        dispatch(startSavingNotes());
    }

    return (
        <Grid
            className='animate__animated animate__fadeIn animate__faster'
            container
            direction='row'
            justifyContent='space-between'
            sx={{ mb: 1 }}
        >
            <Grid item>
                <Typography sx={{ fontSize: 39, fontWeight: "light" }}>{moment(dateString).format('LLLL')}</Typography>
            </Grid>

            <Grid item>
                <Button
                    color='primary'
                    sx={{ padding: 2 }}
                    onClick={onSaveNote}
                    disabled={isSaving}
                >
                    <SaveOutlined sx={{ fontSize: 30, mr: 1 }} />
                    Guardar
                </Button>
            </Grid>

            <Grid container>
                <TextField
                    type='text'
                    variant='filled'
                    fullWidth
                    placeholder="Ingrese un titulo"
                    label='Título'
                    sx={{ border: 'none', mb: 1 }}
                    name='title'
                    value={title}
                    onChange={onInputChange}
                />

                <TextField
                    type='text'
                    variant='filled'
                    fullWidth
                    multiline
                    placeholder="¿Qué sucedió en el día de hoy?"
                    minRows={5}
                    name='body'
                    value={body}
                    onChange={onInputChange}
                />

                <ImageGallery />


            </Grid>

        </Grid>
    )
}
