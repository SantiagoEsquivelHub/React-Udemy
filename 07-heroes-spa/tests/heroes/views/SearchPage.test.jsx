import { fireEvent, render, screen } from "@testing-library/react"
import { MemoryRouter, useNavigate } from "react-router-dom"
import { SearchPage } from "../../../src/heroes/views/SearchPage"

const mockedUseNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUseNavigate
}))

describe('Pruebas en <SearchPage />', () => {

    beforeEach(
        () => jest.clearAllMocks()
    )

    test('debe de mostarse correctamente con los valores por defecto', () => {


        const { container } = render(
            <MemoryRouter>
                <SearchPage />
            </MemoryRouter>
        )

        //screen.debug();
        expect(container).toMatchSnapshot();
    })

    test('debe de mostarse a batman y el input con el valor del queryString', () => {


        render(
            <MemoryRouter initialEntries={['/search?q=batman']}>
                <SearchPage />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox');
        const imgHero = screen.getByRole('img');
        const search_heroe = screen.getByLabelText('search_heroe');

        expect(input.value).toBe('batman');
        expect(search_heroe.style.display).toBe('none');
        expect(imgHero.src).toContain('/assets/heroes/dc-batman.jpg');

    })

    test('debe de mostrar un error si no se encuentra el hero (batman123)', () => {

        render(
            <MemoryRouter initialEntries={['/search?q=batman123']}>
                <SearchPage />
            </MemoryRouter>
        )

        const error_search_heroe = screen.getByLabelText('error_search_heroe');
        expect(error_search_heroe.style.display).toBe('');

    })

    test('debe de llamar el navigate a la pantalla nueva', () => {

        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPage />
            </MemoryRouter>
        )

        const input = screen.getByRole('textbox');
        fireEvent.change(input, { target: { name: 'searchText', value: 'superman' } })

        console.log("input", input.value)


        const form = screen.getByRole('form');
        fireEvent.submit(form);

        expect(mockedUseNavigate).toHaveBeenCalledWith('?q=superman');
    })
})