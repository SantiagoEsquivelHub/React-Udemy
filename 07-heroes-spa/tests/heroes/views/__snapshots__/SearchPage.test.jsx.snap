// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SearchPage /> debe de mostarse correctamente con los valores por defecto 1`] = `
<div>
  <div>
    <h4>
      Search
    </h4>
    <hr />
    <div
      class="row"
    >
      <div
        class="col-5"
      >
        <h4>
          Searching...
        </h4>
        <hr />
        <form
          aria-label="form"
        >
          <input
            autocomplete="off"
            class="form-control mb-2"
            name="searchText"
            placeholder="Search a heroe"
            type="text"
            value=""
          />
          <button
            class="btn btn-outline-primary"
          >
            Search
          </button>
        </form>
      </div>
      <div
        class="col-7"
      >
        <h4>
          Results
        </h4>
        <hr />
        <div
          aria-label="search_heroe"
          class="alert alert-primary animate__fadeIn animate__animated"
        >
          Search a heroe
        </div>
        <div
          aria-label="error_search_heroe"
          class="alert alert-danger animate__fadeIn animate__animated"
          style="display: none;"
        >
          There's no heroe with 
          <b />
        </div>
      </div>
    </div>
  </div>
</div>
`;
